### PR TITLE
fix(cd-service): Renames metrics in cd-servide

### PR DESCRIPTION
**Pull request by diogo-nogueira-freiheit**
The following metrics in the cd-service were renamed:
* `env_lock_count` > `environment_lock_count`
* `app_lock_count` > `application_lock_count`

While the new metrics were created the old ones are still present to avoid a breaking change.
The PR that will remove these old metrics, thus introducing a breaking change is [here](https://github.com/freiheit-com/kuberpult/pull/2095).

The `UpdateDatadogMetrics` function is now independent of the manifest repo.

Ref: SRX-EOXNZ3

### CODE REVIEW

**Comment by diogo-nogueira-freiheit** [comment]
Will create new branch with breaking change and add link to respective PR after review.

File: services/cd-service/pkg/repository/transformer.go [hunk 1]
```diff
@@ -181,43 +199,43 @@ func GaugeDeploymentMetric(_ context.Context, env, app string, timeInMinutes flo
...
+		if err != nil {
+			return fmt.Errorf("failed to read environment from the db: %v", err)
+		}
+
+		// in the future apps might be sorted, but currently they aren't
+		slices.Sort(env.Applications)
+		for _, appName := range env.Applications {
+			GaugeEnvAppLockMetric(ctx, state, transaction, envName, appName)
+
+			// 2024-11-08 17:09:03 +0000 UTC
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/transformer.go:230]: not sure what this comment is trying to tell me
```suggestion
```

File: services/cd-service/pkg/repository/transformer_db_test.go [hunk 2]
```diff
@@ -3691,3 +3694,356 @@ func TestIsCustomErrorRetryable(t *testing.T) {
...
+				makeGauge("app_lock_count", 0, []string{"app:app1", "env:envA"}, 1),
+				makeGauge("application_lock_count", 0, []string{"kuberpult_environment:envA", "kuberpult_application:app1"}, 1),
+				makeGauge("lastDeployed", 0, []string{"kuberpult_application:app1", "kuberpult_environment:envA"}, 1),
+				makeGauge("app_lock_count", 0, []string{"app:app2", "env:envA"}, 1),
+				makeGauge("application_lock_count", 0, []string{"kuberpult_environment:envA", "kuberpult_application:app2"}, 1),
+				makeGauge("lastDeployed", 0, []string{"kuberpult_application:app2", "kuberpult_environment:envA"}, 1),
+
+				// {request_queue_size 0 [] 1}
+				// {env_lock_count 0 [env:envA] 1}
+				// {environment_lock_count 0 [kuberpult_environment:envA] 1}
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/pkg/repository/transformer_db_test.go:3939]:
```suggestion
```

File: docs/metrics.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+## Datadog Metrics
+Kuberpult uploads mulitple metrics to datadog.
+
+### `cd-service` Metrics
+The cd-service uploads the following metrics to datadog:
+* `env_lock_count` - the count of current environment locks, for a given environment; \[❗ deprecated\]
```
Comment (sven-urbanski-freiheit-com) [inline docs/metrics.md:6]: this needs to be updated before merging

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

File: docs/metrics.md (hunk 3 above)
Comment (diogo-nogueira-freiheit) [reply docs/metrics.md:6]: What do you mean by updated? Do you want me to remove the deprecated warning from the docs?

**Pull request merged**